src/sci/impl/fns.cljc

```diff
@@ -242,10 +251,14 @@
         single-arity? (= 1 (count fn-bodies))
         f (if single-arity?
             (fun ctx interpret (first fn-bodies) fn-name macro? false)
-            (let [arities (map #(fun ctx interpret % fn-name macro? true) fn-bodies)]
+            (let [arities (map #(fun ctx interpret % fn-name macro? true) fn-bodies)
```

Comment:
@GreshamDanielStephens Can we maybe collapse all of this processing into one `reduce` for more performance?

``` 
(let [by-arity (reduce (fn [acc body] (assoc acc ...)) fn-bodies)
``` 

We can reserve a special key for the var-args body or return a tuple from the reduce.